sessions/recap-project-5/example-implementation/pages/art-pieces/[slug].js

```diff
@@ -0,0 +1,135 @@
+import Image from "next/image";
+import styled from "styled-components";
+import { useRouter } from "next/router";
+import Link from "next/link";
+import Palette from "../../components/Palette";
+import CommentForm from "../../components/CommentForm";
+import DetailItem from "../../components/DetailItem";
+
+const DetailWrapper = styled.div`
+  display: flex;
+  flex-direction: column;
+  width: 600px;
+  padding-block-end: 5vh;
+`;
+
+const ImageContainer = styled.div`
+  position: relative;
+  width: 100%;
+  height: 30vh;
+`;
+
+const StyledImage = styled(Image)`
+  object-fit: cover;
+`;
+
+const DetailList = styled.dl`
+  display: flex;
+  flex-direction: column;
+  background: #000;
+  color: #fff;
+  margin: 0;
+  padding: 1rem 1rem 2rem 1rem;
+  gap: 1rem;
+`;
+
+const CommentSection = styled.div`
+  display: flex;
+  flex-direction: column;
+  gap: 1rem;
+  margin-block: 3rem;
+`;
+
+const CommentSectionTitle = styled.h2`
+  margin: 0;
+`;
+
+const Comment = styled.p`
+  display: flex;
+  flex-direction: column;
+  gap: 1rem;
+  background: #000;
+  color: #fff;
+  margin: 0;
+  padding: 1rem;
+  border-radius: 20px 20px 0 20px;
+
+  & span:last-child {
+    font-size: 0.7rem;
+    color: grey;
+  }
+`;
+
+const PaletteWrapper = styled.div`
+  display: flex;
+`;
+
+const StyledLink = styled(Link)`
+  width: fit-content;
+  text-decoration: none;
+  color: #fff;
+
+  &:hover {
+    color: lightblue;
+  }
+`;
+
+export default function ArtPieceDetails({
+  pieces,
+  handleSubmitComment,
+  artPiecesInfo,
+}) {
```

Comment:
According to user story 4, this should be a component in order to test it properly.

Reply:
Is this not a component?

Reply:
Yes, it is a component, but you would not test a complete Next.js Page. 

We should have it more or less like this (have not teted the code)

/pages/art-pieces/[slug].js

```jsx
export default function ArtPieceDetailsPage({
  pieces,
  handleSubmitComment,
  artPiecesInfo,
}) {
  const router = useRouter();
  const { slug } = router.query;

  const selectedPiece = pieces.find((piece) => piece.slug === slug);
  const imageInfo = artPiecesInfo.find((infoItem) => infoItem.slug === slug);

  if (!selectedImg) {
    return null;
  }

return (
    <ArtPieceDetails
      image={selectedPiece.image}
      title={selectedPiece.title}
      artist={selectedPiece.artist}
      year={selectedPiece.year}
      genre={selectedPiece.genre}
      comments={imageInfo.comments}
      isFavorite={imageInfo.isFavorite}
    />
  );
});
```

Reply:
Well, component-ish? It's a function in a `pages` file, while a component is a function in a `components` file. Components, however, are not displayed on their own `/componentName` url, while pages are.

And as far as I know, we should not test entire pages; but I'm still not that much into Next.js that I have an answer how to handle and distinguish this.